### ISSUE TITLE
Security vulnerabilities and compliance issue with the docker image

### ISSUE THREAD

**Issue by zaknuces** [amazonlinux/container-images#31]
I run twistlock scan on amazoncorretta image today and following are the findings:
{ "results": [ { "id": "sha256:7e791e7d9b2d2a65e93af34ed393d36029f2e5b63a8c191c4ff28c6b305153b4", "distro": "Amazon Linux 2", "compliances": [ { "title": "Private keys stored in image", "severity": "high", "cause": "Found: /usr/share/doc/pygpgme-0.3/tests/keys/key1.sec, /usr/share/doc/pygpgme-0.3/tests/keys/key2.sec, /usr/share/doc/pygpgme-0.3/tests/keys/signonly.sec" } ], "complianceDistribution": { "critical": 0, "high": 1, "medium": 0, "low": 0, "total": 1 }, "vulnerabilities": [ { "id": "ALAS-2019-1291", "status": "fixed in 2.7.16-3.amzn2.0.1", "description": "CVE-2019-9948", "severity": "medium", "packageName": "python", "packageVersion": "2.7.16-2.amzn2.0.1", "link": "https://alas.aws.amazon.com/AL2/ALAS-2019-1291.html", "riskFactors": { "Has fix": {}, "Medium severity": {} } }, { "id": "ALAS-2019-1305", "status": "fixed in 3.44.0-4.amzn2.0.2", "description": "CVE-2018-0495 CVE-2018-12404", "severity": "medium", "packageName": "nss-tools", "packageVersion": "3.36.0-7.amzn2", "link": "https://alas.aws.amazon.com/AL2/ALAS-2019-1305.html", "riskFactors": { "Has fix": {}, "Medium severity": {} } }, { "id": "ALAS-2019-1301", "status": "fixed in 2.9.1-6.amzn2.3.3", "description": "CVE-2016-4658 CVE-2017-16931", "severity": "medium", "packageName": "libxml2", "packageVersion": "2.9.1-6.amzn2.3.2", "link": "https://alas.aws.amazon.com/AL2/ALAS-2019-1301.html", "riskFactors": { "Has fix": {}, "Medium severity": {} } }, { "id": "ALAS-2019-1291", "status": "fixed in 2.7.16-3.amzn2.0.1", "description": "CVE-2019-9948", "severity": "medium", "packageName": "python-libs", "packageVersion": "2.7.16-2.amzn2.0.1", "link": "https://alas.aws.amazon.com/AL2/ALAS-2019-1291.html", "riskFactors": { "Has fix": {}, "Medium severity": {} } }, { "id": "ALAS-2019-1305", "status": "fixed in 3.44.0-4.amzn2.0.2", "description": "CVE-2018-0495 CVE-2018-12404", "severity": "medium", "packageName": "nss", "packageVersion": "3.36.0-7.amzn2", "link": "https://alas.aws.amazon.com/AL2/ALAS-2019-1305.html", "riskFactors": { "Has fix": {}, "Medium severity": {} } }, { "id": "ALAS-2019-1289", "status": "fixed in 2.56.1-4.amzn2", "description": "CVE-2019-12450", "severity": "medium", "packageName": "glib2", "packageVersion": "2.54.2-2.amzn2", "link": "https://alas.aws.amazon.com/AL2/ALAS-2019-1289.html", "riskFactors": { "Has fix": {}, "Medium severity": {} } }, { "id": "ALAS-2019-1305", "status": "fixed in 3.44.0-4.amzn2.0.2", "description": "CVE-2018-0495 CVE-2018-12404", "severity": "medium", "packageName": "nss-sysinit", "packageVersion": "3.36.0-7.amzn2", "link": "https://alas.aws.amazon.com/AL2/ALAS-2019-1305.html", "riskFactors": { "Has fix": {}, "Medium severity": {} } }, { "id": "ALAS-2019-1302", "status": "fixed in 6.0-8.20170212.amzn2.1.3", "description": "CVE-2017-10684 CVE-2017-10685 CVE-2017-11112 CVE-2017-11113", "severity": "medium", "packageName": "ncurses-libs", "packageVersion": "6.0-8.20170212.amzn2.1.2", "link": "https://alas.aws.amazon.com/AL2/ALAS-2019-1302.html", "riskFactors": { "Has fix": {}, "Medium severity": {} } }, { "id": "ALAS-2019-1302", "status": "fixed in 6.0-8.20170212.amzn2.1.3", "description": "CVE-2017-10684 CVE-2017-10685 CVE-2017-11112 CVE-2017-11113", "severity": "medium", "packageName": "ncurses", "packageVersion": "6.0-8.20170212.amzn2.1.2", "link": "https://alas.aws.amazon.com/AL2/ALAS-2019-1302.html", "riskFactors": { "Has fix": {}, "Medium severity": {} } }, { "id": "ALAS-2019-1303", "status": "fixed in 1.4.3-12.amzn2.2.2", "description": "CVE-2019-3862", "severity": "medium", "packageName": "libssh2", "packageVersion": "1.4.3-12.amzn2.2.1", "link": "https://alas.aws.amazon.com/AL2/ALAS-2019-1303.html", "riskFactors": { "Has fix": {}, "Medium severity": {} } } ], "vulnerabilityDistribution": { "critical": 0, "high": 0, "medium": 10, "low": 0, "total": 10 } } ] }

All the security issues are fixable by the underlying image. More so, can we also look at compliance issue, it is complaining about keys in the test package. If we can remove them that would be great.

**Comment by iliana** [comment]
Tracking an image update in #30; I'm opening an issue internally to look at the compliance issue (sample keys in `%doc` of pygpgme).

**Comment by iliana** [comment]
Closed #30 as I've just pushed an image, but keeping this open to track the pygpgme issue.

**Comment by zaknuces** [comment]
@iliana this is the scan results of the latest image.

```
"results": [
		{
			"id": "sha256:5dc6819f98d7cb1fcee9db5fab55a8a9016e3111c95f90bdbf651d1094dcb68e",
			"distro": "Amazon Linux 2",
			"compliances": [
				{
					"title": "Private keys stored in image",
					"severity": "high",
					"cause": "Found: /usr/share/doc/pygpgme-0.3/tests/keys/key1.sec, /usr/share/doc/pygpgme-0.3/tests/keys/key2.sec, /usr/share/doc/pygpgme-0.3/tests/keys/signonly.sec"
				}
			],
			"complianceDistribution": {
				"critical": 0,
				"high": 1,
				"medium": 0,
				"low": 0,
				"total": 1
			},
			"vulnerabilityDistribution": {
				"critical": 0,
				"high": 0,
				"medium": 0,
				"low": 0,
				"total": 0
			}
		}
	]
```

No security vulnerabilities detected. Only compliance issue related to pygpgme.
